taipy/event/event_consumer.py

```diff
@@ -0,0 +1,1237 @@
+                    event_consumer = GuiEventConsumer()
+                    event_consumer.broadcast_on_scenario_created(callback=scenario_created, scenario_config="my_cfg")
+                    event_consumer.start()
+                    ...
+                ```
+
+        Arguments:
+            callback (callable):The callback to be executed for each state when
+                a scenario creation event occurs.
+                ```python
+                def on_event_received(state: State, event: Event, scenario: Scenario):
+                    ...
+                ```
+                The callback takes the state, the event, and the scenario as arguments.
+                Optionally, the callback can accept extra arguments (see the
+                `callback_args` argument).
+            callback_args (List[AnyOf]): The extra arguments to be passed to the callback
+                function in addition to the state, the event, and the scenario.
+            scenario_config (Union[str, ScenarioConfig, List, None]): The
+                optional scenario configuration ids or scenario configurations
+                for which the callback is registered. If None, the callback is registered
+                for all scenario configurations.
+
+        Returns:
+            GuiEventConsumer: The current instance of the `GuiEventConsumer` service.
+        """
+        return self.__on_scenario_created(
+            callback=callback,
+            callback_args=callback_args,
+            scenario_config=scenario_config,
+            broadcast=True,
+        )
+
+    def __on_scenario_created(self,
+                              callback: Callable,
+                              callback_args: Optional[List] = None,
+                              scenario_config: Union[str, ScenarioConfig, List, None] = None,
+                              broadcast: bool = False,
+                              ) -> 'GuiEventConsumer':
+        scenario_config = self.__format_configs_parameter(ScenarioConfig, scenario_config)
+
+        def _filter(event: Event) -> bool:
+            import taipy as tp
+
+            sc = tp.get(event.entity_id)
+            if scenario_config and sc.config_id not in scenario_config:
+                return False
+            event.metadata["predefined_args"] = [sc]
+            return True
+
+        self.__on_event(callback=callback,
+                        callback_args=callback_args,
+                        entity_type=EventEntityType.SCENARIO,
+                        operation=EventOperation.CREATION,
+                        filter=_filter,
+                        broadcast=broadcast)
+        return self
+
+    def on_scenario_deleted(self,
+                            callback: Callable,
+                            callback_args: Optional[List] = None,
+                            scenario_config: Union[str, ScenarioConfig, List, None] = None,
+                            ) -> 'GuiEventConsumer':
```

Comment:
I'm surprise that ruff is not fixing this

```suggestion
                            ) -> "GuiEventConsumer":
```